src/main/kotlin/com/cognifide/gradle/aem/instance/InstancePlugin.kt

```diff
@@ -32,44 +32,54 @@ class InstancePlugin : AemPlugin() {
 
             register<InstanceDown>(InstanceDown.NAME)
             register<InstanceUp>(InstanceUp.NAME) {
-                dependsOn(InstanceCreate.NAME)
                 mustRunAfter(InstanceDown.NAME, InstanceDestroy.NAME)
             }
             register<InstanceRestart>(InstanceRestart.NAME) {
                 dependsOn(InstanceDown.NAME, InstanceUp.NAME)
             }
-            register<InstanceCreate>(InstanceCreate.NAME) {
+            register<InstanceCreateOnly>(InstanceCreateOnly.NAME) {
                 dependsOn(Resolve.NAME)
                 mustRunAfter(InstanceDestroy.NAME)
             }
+            register<InstanceCreateAndUp>(InstanceCreateAndUp.NAME) {
```

Comment:
previous semantic was so nice; I don't know how you will convince me to go with that :)

Reply:
I will move up logic to tasks